Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -614,6 +614,7 @@ namespace AZ
 
 #if !defined(_RELEASE)
         m_budgetTracker.Init();
+        m_statisticalProfiler.reset(new AZ::Statistics::StatisticalProfilerProxy);
```

Comment:
StatisticalProfiler needs to be initialized here since budgets are lazy initialized on first use and they could be called very early through the profile macros. The statistical profiler is used when a Budget instance is created.